File: resources/assets/admin/src/api/banner.js
import request from '@/utils/request'

export function list(query) {
  return request({
    url: '/v1/banner/',
    method: 'get',
    params: query
  })
}

export function create(data) {
  return request({
    url: '/v1/banner/',
    method: 'post',
    data
  })
}

export function update(id, data) {
  return request({
    url: '/v1/banner/' + id,
    method: 'put',
    data
  })
}

export function show(id) {
  return request({
    url: '/v1/banner/' + id,
    method: 'get'
  })
}

export function del(id) {
  return request({
    url: '/v1/banner/' + id,
    method: 'delete'
  })
}

// 删除banner组
export function delBanners(ids) {
  return request({
    url: '/v1/banner/',
    method: 'delete',
    data: {
      ids: ids
    }
  })
}

File: app/service/banner.go
package service

import (
	"fagin/app/models/banner"
	"fagin/pkg/db"
	"github.com/gin-gonic/gin"
)

type bannerService struct{}

var Banner bannerService

func (*bannerService) Index(params gin.H, columns []string, with gin.H, p *db.Paginator) ([]banner.Banner, error) {
	var banners []banner.Banner
	err := banner.NewDao().Query(params, columns, with).Paginate(&banners, p)
	return banners, err
}

func (*bannerService) Show(id uint, columns []string) (*banner.Banner, error) {
	b := banner.New()
	err := b.Dao().FindById(id, columns)
	return b, err
}

func (*bannerService) Create(m *banner.Banner) error {
	return banner.NewDao().Create(m)
}

func (*bannerService) Update(id uint, data gin.H) error {
	return banner.NewDao().Update(id, data)
}

func (*bannerService) Delete(id uint) error {
	return banner.NewDao().Destroy(id)
}

func (*bannerService) DeleteBanners(ids []uint) error {
	return banner.NewDao().Deletes(ids)
}
